torch/csrc/autograd/FunctionsManual.h

```diff
@@ -38,6 +38,11 @@ Tensor toNonOptFwGrad(const c10::optional<Tensor>& t);
 Tensor toNonOptPrimal(const c10::optional<Tensor>& t);
 Tensor toNonOptTensor(const c10::optional<Tensor>& t);
 
+inline c10::optional<Tensor> wrap_opt_if(const Tensor& t, const bool cond) {
+  using OptTensor = c10::optional<Tensor>;
+  return cond ? OptTensor(t) : static_cast<OptTensor>(c10::nullopt);
+}
+
```

Comment:
For context: this one is used in the next PR up in the stack for `sparse_sampled_addmm_backward`.

Reply:
If codegen did its job correctly, we would always get an undefined tensor t here right? Maybe we can assert for that here.

Reply:
No, not really, unfortunately. This code is being run at backward compute. But we can assert inside backward implementations for sure.

Reply:
oh whoops, good point.